File: apps/web/src/api/health.ts
import http from "./http";

export interface HealthResponse {
  status: string;
  data: {
    service: string;
    version: string;
    time: string;
  };
}

export async function fetchHealth(): Promise<HealthResponse> {
  const { data } = await http.get<HealthResponse>("/api/v1/health");
  return data;
}

File: apps/api/app/api/v1/endpoints/health.py
from datetime import datetime, timezone

from fastapi import APIRouter, Depends

from app.core.config import Settings, get_settings

router = APIRouter()


@router.get("/health", summary="Health check")
def health_check(settings: Settings = Depends(get_settings)) -> dict:
    now = datetime.now(timezone.utc).isoformat()
    return {
        "status": "ok",
        "data": {
            "service": settings.project_name,
            "version": settings.version,
            "time": now,
        },
    }
